src/sinks/amqp/config.rs

```diff
@@ -0,0 +1,98 @@
+//! Configuration functionality for the AMQP sink.
+use crate::{
+    amqp::AMQPConfig,
+    codecs::EncodingConfig,
+    config::{DataType, GenerateConfig, Input, SinkConfig, SinkContext},
+    sinks::{Healthcheck, VectorSink},
+};
+use codecs::TextSerializerConfig;
+use futures::FutureExt;
+use std::sync::Arc;
+use vector_config::configurable_component;
+use vector_core::config::AcknowledgementsConfig;
+
+use super::sink::AMQPSink;
+
+/// Configuration for the `amqp` sink.
+///
+/// Supports AMQP version 0.9.1
+#[configurable_component(sink("amqp"))]
+#[derive(Clone, Debug)]
+pub struct AMQPSinkConfig {
+    /// The exchange to publish messages to.
+    pub(crate) exchange: String,
+
+    /// Template used to generate a routing key which corresponds to a queue binding.
+    // TODO: We will eventually be able to add metadata on a per-field basis, such that we can add metadata for marking
+    // this field as being capable of using Vector's templating syntax.
+    pub(crate) routing_key: Option<String>,
```

Comment:
```suggestion
    /// The name of the field that contains the routing_key of the event.
    #[configurable(metadata(templateable))]
    pub(crate) routing_key: Option<Template>,
```
Though I'm not sure why this should be templateable at all, this seems to be a log_namespace/schema style option that would be the comparable to [host_key](https://vector.dev/docs/reference/configuration/sources/internal_metrics/#tags.host_key)

Reply:
Not sure I understand what you mean. The value of the routing key can be used by Rabbit to determine which queues the message should be posted on, so I can see the value of providing the most flexibility in setting this value. Not sure how this compares to host_key?

Reply:
🤦 sorry - mixed up between the source and sink here. Ignore my additional comment, but the annotation can still be used instead of the TODO